lua/entities/gmod_wire_expression2/core/custom/pewpewfunctions.lua

```diff
@@ -455,4 +455,17 @@ e2function entity pewCreate( string bullet, string model, vector pos, angle ang
 	return create(self.player, bullet, model, pos, ang)
 end
 
+
+__e2setcost(0)
+
+e2function number pewTimeLastFired( entity cannon )
+    if not IsValid(cannon) then return -1 end
+    return CurTime() - cannon.LastFired
+end
+
+e2function number pewTimeCanFire( entity cannon )
+    if not IsValid(cannon) then return 0 end
+    return  ( cannon.Bullet.Reloadtime + cannon.LastFired ) - CurTime() 
```

Comment:
```
if not IsValid(cannon) then return -1 end
return math.max(0, ( cannon.Bullet.Reloadtime + cannon.LastFired ) - CurTime())
```

Reply:
i think you might have a point with this one chief.